services/server/src/server/services/StorageService.ts

```diff
@@ -0,0 +1,62 @@
+import { Match, CheckedContract } from "@ethereum-sourcify/lib-sourcify";
+import { logger } from "../../common/loggerLoki";
+import {
+  IpfsRepositoryService,
+  IpfsRepositoryServiceOptions,
+} from "./storageServices/IpfsRepositoryService";
+import {
+  AllianceDatabaseService,
+  AllianceDatabaseServiceOptions,
+} from "./storageServices/AllianceDatabaseService";
+
+export interface IStorageService {
+  init(): Promise<boolean>;
+  storeMatch(contract: CheckedContract, match: Match): Promise<void | Match>;
```

Comment:
I don't think they are mandatory, but we can add them as optional functions

Reply:
Shouldn't storing (`storeMatch`) and retrieving (`checkByChainAndAddress` and `checkAllByChainAndAddress`) be required?

Reply:
`storeMatch` yes because every Storage service must, at least, store something. But, for example, the Alliance Database doesn't implement the method `checkByChainAndAddress` because we never check directly with the Alliance Database if the contracts are available over there.